CONTRIBUTING.md

```diff
@@ -89,3 +89,83 @@ Reference [pre-commit's installation instructions](https://pre-commit.com/#insta
+
+Please refer to the Docker documentation for other common container actions.
+
+Development on Houston or other Codex components should be done by testing new code against the full application in these running docker containers. Running tests or migrations outside the docker-compose environment is an extra tool at your disposal but can be unpredictable and is not considered complete.
+
+### Mounted Directories
+
+#### Houston
+The docker-compose arrangement will attempt to mount local directories for development purposes. For Houston, this is the root repository directory. 
+
+If you create or modify a file in the local Houston repository you will be able to see the changes reflected when you `docker exec` inside the Houston container.
+
+This does not go the other way. Changes made inside the container reside only there.
+
+#### Frontend
+The docker-compose.yml file also mounts a `_frontend` directory for the front end application. Since we do not build the front end locally by default, this does not map to any location.
+If you want to build the front end, use the command `invoke dependencies.install-frontend-ui`. This will create a `_frontend` folder in the houston repo that contains files which 
+can be modified, and the changes will be reflected in your running `codex_dev-frontend` container.
+
+If you want to change the mountpoint to a different directory for your locally cloned Codex frontend repository to make changes and commits easier, you can change it
+in the `deploy/codex/docker-compose.yml` by altering the `dev-frontent` volume mapping `- ../../_frontend:/code` to a directory outside your Houston repository.
+
+More details about Codex front end contribution are outside the scope of this README but can be found here: [**codex-frontend**](https://github.com/WildMeOrg/codex-frontend)
+
+#### EDM
+The EDM is a compiled Java application, and no volume mapping solution to a running Docker container is available at this time. 
+
+### Testing
+
+New Houston code must be tested with `pytest`. If dependencies are set up correctly an initial testing run can be done outside the docker container 
+with the `pytest` command at the root level of the repository.
+
+To fully test, `docker exec` into the Houston container and run tests there by setting `FLASK_CONFIG= pytest -s -x` on the command line. 
+
+Both of these methods can target a specific app module by altering the command to something like this:
+
+`pytest tests/modules/[MODULE NAME]` or `FLASK_CONFIG= pytest tests/modules/[MODULE NAME] -s -x`
+
+### Migrations
+
+If your modifications creates a database migration, the `invoke` tasks to perform the migration must be tested inside the Houston docker container.
+
+### Rebuilding
+
+In the process of contributing you will want to sync up with the latest Houston/Codex code and not be working against Docker leftovers. 
+
+These invoke commands will assist you in a clean start: 
```

Comment:
Would be helpful to have some indication of when these should be run. 
I current run these type of things when stuff stops working but that feels like the wrong way to do it.